titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/client/DefaultDirectKubeApiServerIntegrator.java

```diff
@@ -149,6 +154,28 @@ public boolean isReadyForScheduling() {
         return kubeApiFacade.isReadyForScheduling();
     }
 
+    public List<ContainerState> getPodStatus(String taskId) {
+        if (getPods().containsKey(taskId)) {
```

Comment:
instead of using `getPods` which creates a new map each time, you can use the `pods` private variable directly.